2024-12-07 21:22:47,303 ERROR: Exception on / [GET] [in D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\routes.py", line 73, in index
    return render_template("index.html", title='Home', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\PROJECTS\FlaskHabr\app\templates\index.html", line 2, in top-level template code
    {% import "bootstrap_wtf.html" as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\base.html", line 59, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\index.html", line 5, in block 'content'
    <h1>{{ _('Hi, %(username)s!', username=current_user.username) }}</h1>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\ext.py", line 168, in _gettext_alias
    return __context.call(__context.resolve("gettext"), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\ext.py", line 174, in gettext
    rv = __context.call(func, __string)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 166, in <lambda>
    gettext=lambda s: get_translations().ugettext(s),
                      ^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 246, in get_translations
    return get_domain().get_translations()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 609, in get_translations
    locale = get_locale()
             ^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 263, in get_locale
    rv = babel.locale_selector()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\__init__.py", line 17, in get_locale
    return request.accept_language.best_match(app.config['LANGUAGES'])
           ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Request' object has no attribute 'accept_language'. Did you mean: 'accept_languages'?
2024-12-07 21:22:55,264 ERROR: Exception on /index [GET] [in D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\routes.py", line 73, in index
    return render_template("index.html", title='Home', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\PROJECTS\FlaskHabr\app\templates\index.html", line 2, in top-level template code
    {% import "bootstrap_wtf.html" as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\base.html", line 59, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\templates\index.html", line 5, in block 'content'
    <h1>{{ _('Hi, %(username)s!', username=current_user.username) }}</h1>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\ext.py", line 168, in _gettext_alias
    return __context.call(__context.resolve("gettext"), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\jinja2\ext.py", line 174, in gettext
    rv = __context.call(func, __string)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 166, in <lambda>
    gettext=lambda s: get_translations().ugettext(s),
                      ^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 246, in get_translations
    return get_domain().get_translations()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 609, in get_translations
    locale = get_locale()
             ^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\env\Lib\site-packages\flask_babel\__init__.py", line 263, in get_locale
    rv = babel.locale_selector()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\PROJECTS\FlaskHabr\app\__init__.py", line 17, in get_locale
    return request.accept_language.best_match(app.config['LANGUAGES'])
           ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Request' object has no attribute 'accept_language'. Did you mean: 'accept_languages'?
